packages/application-shell/src/components/app-bar/app-bar.js

```diff
@@ -31,31 +31,40 @@ const AppBar = props => (
           <Spacings.Inline alignItems="center">
             {/* This node is used by a react portal */}
             <div id="locale-switcher" />
-            {(() => {
+            {do {
+              const previouslyUsedProjectKey = selectProjectKeyFromLocalStorage();
               if (!props.user) {
-                return <LoadingPlaceholder shape="rect" size="s" />;
+                <LoadingPlaceholder shape="rect" size="s" />;
               }
               // The `<ProjectSwitcher>` should be rendered only if the
-              // user is fetched and the user has projects.
-              if (props.user.projects.total > 0) {
-                return (
-                  <ProjectSwitcher
-                    // In this case it's not necessary to check if the `projectKey` param
-                    // is included in the list of projects. In such case
-                    // the dropdown will still be rendered but no project will be selected.
-                    // This is fine becase the user has still the possibility to "switch"
-                    // to a project.
-                    projectKey={
-                      selectProjectKeyFromUrl() ||
-                      selectProjectKeyFromLocalStorage() ||
-                      props.user.defaultProjectKey
-                    }
-                    total={props.user.projects.total}
-                  />
-                );
-              }
-              return null;
-            })()}
+              // user is fetched and the user has projects while the app runs in an project context.
+              else if (props.user.projects.total > 0 && props.projectKeyFromUrl)
+                <ProjectSwitcher
+                  // In this case it's not necessary to check if the `projectKey` param
+                  // is included in the list of projects. In such case
+                  // the dropdown will still be rendered but no project will be selected.
+                  // This is fine becase the user has still the possibility to "switch"
+                  // to a project.
+                  projectKey={
+                    props.projectKeyFromUrl ||
+                    previouslyUsedProjectKey ||
+                    props.user.defaultProjectKey
+                  }
+                  total={props.user.projects.total}
+                />;
+              else
+                <Text.Detail>
+                  <Link
+                    to={`/${previouslyUsedProjectKey ||
```

Comment:
I tried to work `onelineTrim` here but it's really mostly the condition which creates a bit of a strange formatting here.